common/src/truffle.rs

```diff
@@ -103,3 +104,32 @@ mod tests {
         }
     }
 }
+
+/// Deserialization implementation for compatibility with ABIs produced with
+/// solc v0.6. This is a known `ethabi` issue with an open PR for a fix:
+/// https://github.com/openethereum/ethabi/issues/185
+/// https://github.com/openethereum/ethabi/pull/187
+mod compat_solc_v0_6 {
+    use super::*;
+    use serde::de::{self, Deserializer};
+    use serde_json::{json, Value};
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Abi, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let mut raw_abi: Vec<Value> = Deserialize::deserialize(deserializer)?;
+        let receive = json!("receive");
+        for entry in raw_abi.iter_mut() {
+            if let Value::Object(obj) = entry {
+                if let Some(kind) = obj.get_mut("type") {
+                    if kind == &receive {
+                        *kind = json!("fallback");
```

Comment:
Is this correct when there already is a fallback function?

Reply:
`receive` is a payable fallback function from what I understand. Currently `ethabi` doesn't differentiate between payable and non-payable fallback functions and allows more than one in the ABI, so this should be fine.

Reply:
If it allows more than one then it seems fine.